src/test/java/com/synopsys/integration/alert/database/api/ProcessingJobAccessorTestIT.java

```diff
@@ -0,0 +1,240 @@
+import com.synopsys.integration.alert.util.AlertIntegrationTest;
+import com.synopsys.integration.blackduck.api.generated.enumeration.VulnerabilitySeverityType;
+import com.synopsys.integration.blackduck.api.manual.component.AffectedProjectVersion;
+import com.synopsys.integration.blackduck.api.manual.component.VulnerabilityNotificationContent;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@AlertIntegrationTest
+public class ProcessingJobAccessorTestIT {
+    private static final List<UUID> CREATED_JOBS = new LinkedList<>();
+    private static final String PROJECT_NAME_1 = "testProject1";
+    private static final String PROJECT_NAME_2 = "testProject2";
+    private Long providerConfigId;
+
+    @Autowired
+    public JobAccessor jobAccessor;
+    @Autowired
+    public ProcessingJobAccessor processingJobAccessor;
+    @Autowired
+    public ConfigurationAccessor configurationAccessor;
+    @Autowired
+    public DistributionJobRepository distributionJobRepository;
+
+    @BeforeEach
+    public void createProvider() {
+        ConfigurationFieldModel providerConfigEnabled = ConfigurationFieldModel.create(ProviderDescriptor.KEY_PROVIDER_CONFIG_ENABLED);
+        providerConfigEnabled.setFieldValue("true");
+        ConfigurationFieldModel providerConfigName = ConfigurationFieldModel.create(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME);
+        providerConfigName.setFieldValue("My Black Duck Config");
+
+        ConfigurationFieldModel blackduckUrl = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_URL);
+        blackduckUrl.setFieldValue("https://a-blackduck-server");
+        ConfigurationFieldModel blackduckApiKey = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_API_KEY);
+        blackduckApiKey.setFieldValue("123456789012345678901234567890123456789012345678901234567890");
+        ConfigurationFieldModel blackduckTimeout = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_TIMEOUT);
+        blackduckTimeout.setFieldValue("300");
+
+        List<ConfigurationFieldModel> providerConfigFields = List.of(providerConfigEnabled, providerConfigName, blackduckUrl, blackduckApiKey, blackduckTimeout);
+        providerConfigId = configurationAccessor.createConfiguration(new BlackDuckProviderKey(), ConfigContextEnum.GLOBAL, providerConfigFields).getConfigurationId();
+    }
+
+    @AfterEach
+    public void removeCreatedJobsIfExist() {
+        CREATED_JOBS.forEach(jobAccessor::deleteJob);
+        CREATED_JOBS.clear();
+        configurationAccessor.deleteConfiguration(providerConfigId);
+    }
+
+    @Test
+    public void testMatchingEnabledJobsUniqueJobsPerPage() throws InterruptedException {
+        int expectedNumOfJobs = 1000;
+        Set<UUID> previousJobIdSet = new HashSet<>();
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
```

Comment:
createJobs taken from the JobNotificationMapperTestIT class to be able to quickly reproduce the problem.